dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java

```diff
@@ -242,7 +245,27 @@ public App getApp(String appName) {
 
   @Override
   public List<WebModule> getMenu(String contextPath) {
-    return getAccessibleAppMenu(contextPath);
+    List<WebModule> modules = getAccessibleAppMenu(contextPath);
+
+    // Apply user-defined favorite apps order
+    User currentUser = userService.getUserByUsername(CurrentUserUtil.getCurrentUsername());
+
+    if (currentUser != null && currentUser.getApps() != null && !currentUser.getApps().isEmpty()) {
+      final List<String> userApps = new ArrayList<>(currentUser.getApps());
+
+      modules.sort(
```

Comment:
I think what you want is `Comparator.comparing(User:getName).nullsLast()` or something like that, can't remember the exact names of how you control nulls but there is something for it.

Reply:
I think this is a little different because it's looking up the name in a different ordered list and placing any not-found at the end.  This is exactly copied from the previous implementation in GetModulesAction (now removed), definitely could be improved in the future but that has been working for a long time so I think the lowest-risk thing to do is just keep it as-is for now and refactor in the future.